src/OpenLoco/src/Ui/Windows/Vehicle.cpp

```diff
@@ -1137,48 +1173,106 @@ namespace OpenLoco::Ui::Windows::Vehicle
         static void onResize(Window& self)
         {
             Common::setCaptionEnableState(&self);
-            self.setSize(kMinWindowSize, kMaxWindowSize);
+            if (CompanyManager::getControllingId() == self.owner)
+            {
+                self.setSize(kMinWindowSizeWithPaintEnabled, kMaxWindowSize);
+            }
+            else
+            {
+                self.setSize(kMinWindowSize, kMaxWindowSize);
+            }
         }
 
         static void onMouseDown(Window& self, const WidgetIndex_t widgetIndex, [[maybe_unused]] const WidgetId id)
         {
-            if (widgetIndex != widx::buildNew)
+            if (widgetIndex == widx::buildNew)
+            {
+                Dropdown::add(0, StringIds::dropdown_stringid, StringIds::dropdown_modify_vehicle);
+                Dropdown::add(1, StringIds::dropdown_stringid, StringIds::dropdown_clone_vehicle);
+
+                auto& widget = self.widgets[widx::buildNew];
+                Dropdown::showText(
+                    self.x + widget.left,
+                    self.y + widget.top,
+                    widget.width(),
+                    widget.height(),
+                    self.getColour(WindowColour::secondary),
+                    2,
+                    0);
+
+                Dropdown::setItemSelected(0);
+                Dropdown::setHighlightedItem(0);
+                return;
+            }
+            if (widgetIndex == widx::paintColourPrimary || widgetIndex == widx::paintColourSecondary)
             {
+                auto availableColours = 0x7FFFFFFF;
+                Colour selectedColour = ImageId::fromUInt32(self.widgets[widgetIndex].image).getPrimary();
+                Dropdown::showColour(&self, &self.widgets[widgetIndex], availableColours, selectedColour, self.getColour(WindowColour::secondary));
                 return;
             }
+            if (widgetIndex == widx::paintBrush)
+            {
+                Dropdown::add(0, StringIds::dropdown_stringid, StringIds::vehicleRepaintTool);
+                Dropdown::add(1, StringIds::dropdown_stringid, StringIds::vehicleRepaintEntireVehicle);
```

Comment:
I don't think we need this dropdown with this option to repaint the entire vehicle, as this behaviour is covered by Shift+Click. Having the dropdown seems convoluted and redundant. I think the tooltip for the paint button should explain the usage of the tool - "Click to paint a car. Shift+Click to paint the train. Ctrl+Click to paint a component of a car."

Reply:
Reading back, I think the reasoning behind having both was that you don't (easily) have access to these modifier keys on touch interfaces. I'm fine keeping both, personally.

Reply:
Noting that touch users don't have access to Ctrl modifier either, so they won't be able to paint by component. I'm happy to check this in however and we can find a way to enable that later (another dropdown option?)